watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -185,18 +286,30 @@ def add_costing(m):
         ),
         doc="Revenue from recovering saline brine/ NF permeate",
     )
-
-    m.fs.dye_disposal_cost = Expression(
-        expr=(
-            m.fs.zo_costing.utilization_factor
-            * m.fs.zo_costing.dye_disposal_cost
-            * pyunits.convert(
-                m.fs.dye_retentate.properties[0].flow_vol,
-                to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
-            )
-        ),
-        doc="Cost of disposing of dye waste",
-    )
+    if hasattr(m.fs, "gac"):
+        m.fs.dye_disposal_cost = Expression(
+            expr=(
+                m.fs.zo_costing.utilization_factor
+                * m.fs.zo_costing.dye_disposal_cost
+                * pyunits.convert(
+                    m.fs.adsorbed_dye.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+                )
+            ),
+            doc="Cost of disposing of dye waste",
+        )
+    else:
+        m.fs.dye_disposal_cost = Expression(
+            expr=(
+                m.fs.zo_costing.utilization_factor
+                * m.fs.zo_costing.dye_disposal_cost
+                * pyunits.convert(
+                    m.fs.dye_retentate.properties[0].flow_vol,
+                    to_units=pyunits.m**3 / m.fs.zo_costing.base_period,
+                )
+            ),
+            doc="Cost of disposing of dye waste",
```

Comment:
In subsequent PR, we can probably handle this with a Product block for the disposal stream. Then, we wouldn't need to worry about the logic of whether gac is included.